pgctl/poll_ready.py

```diff
@@ -88,14 +113,22 @@ def main():
         timeout = getval('timeout-ready', 'PGCTL_TIMEOUT', '2.0')
         poll_ready = getval('poll-ready', 'PGCTL_POLL', '0.15')
         poll_down = getval('poll-down', 'PGCTL_POLL', '10.0')
-        # this subprocess listens for the s6 down event: http://skarnet.org/software/s6/s6-supervise.html
-        from .subprocess import Popen
-        down_event = Popen(
-            ('s6-ftrig-wait', 'event', 'd'),
-            stdout=open(os.devnull, 'w'),  # this prints 'd' otherwise
-        )
-
-        return pgctl_poll_ready(down_event, notification_fd, timeout, poll_ready, poll_down)
+
+        # Don't reuse an old FIFO
+        if os.path.exists(DOWN_FIFO_PATH):
+            os.remove(DOWN_FIFO_PATH)
+
+        # FIFO listens for the s6 down event
+        #
+        # Even though the FIFO is effectively RO, it is opened as RW because
+        # opening as RO blocks until the other side of the FIFO is opened.
+        os.mkfifo(DOWN_FIFO_PATH)
```

Comment:
I believe that isn't possible. The control flow should be pgctl -> S6 -> poll ready, and S6 creates that fifodir.